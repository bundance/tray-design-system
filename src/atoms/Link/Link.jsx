import React from "react";
import styled from "styled-components";
import { system } from "styled-system";

system({
  prop: "textDecoration",
  cssProperty: "textDecoration",
});

// Create a styled Link
export const StyledLink = styled.a`
  ${system({
    textDecoration: true,
    fontFamily: true,
    fontSize: {
      property: "fontSize",
      scale: "fontSizes",
    },
    color: {
      property: "color",
      scale: "colors",
    },
    backgroundColor: {
      property: "backgroundColor",
      scale: "background",
    },
    fontWeight: {
      property: "fontWeight",
      scale: "fontWeights",
    },
    margin: {
      property: "margin",
      scale: "space",
    },
    marginLeft: {
      property: "margin",
      scale: "space",
    },
    marginRight: {
      property: "margin",
      scale: "space",
    },
    padding: {
      property: "padding",
      scale: "space",
    },
    textAlign: true,
  })}
`;

// The Link component itself is a styled link wih a custom onClick event handler passed
const Link = (props) => (
  <StyledLink
    {...props}
    onClick={(e) => {
      e.preventDefault();
      props.onClick(e);
    }}
  />
);

export default Link;
