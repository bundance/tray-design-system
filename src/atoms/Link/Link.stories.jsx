import React from "react";
import { storiesOf } from "@storybook/react";

import Link from "./Link";

storiesOf("Molecules/Link", module)
  .add("Default", () => <Link href="#">Click here</Link>)
  .add("Styled", () => (
    <Link
      href="#"
      fontFamily="Roboto"
      color="red"
      backgroundColor="blue"
      textDecoration="none"
      fontSize={10}
    >
      Click here
    </Link>
  ));
