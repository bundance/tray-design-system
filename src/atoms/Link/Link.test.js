import React from "react";

import { fireEvent, render } from "@testing-library/react";
import Link from "./Link";

test("it renders correctlyy", () => {
  const { container } = render(<Link />);
  expect(container.firstChild).toMatchSnapshot();
});

test("it calls callback function on click", () => {
  const mockCallback = jest.fn();
  const { getByText } = render(<Link onClick={mockCallback}>link</Link>);

  expect(mockCallback).not.toHaveBeenCalled();

  const link = getByText("link");
  fireEvent(
    link,
    new MouseEvent("click", {
      bubbles: true,
      cancelable: true,
    })
  );
  expect(mockCallback).toHaveBeenCalled();
});
