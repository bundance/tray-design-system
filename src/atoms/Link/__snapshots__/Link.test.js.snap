// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctlyy 1`] = `
<a
  class="sc-AxjAm hyNcgC"
/>
`;
